doc/tests/block.rst

```diff
@@ -0,0 +1,13 @@
+``block``
+=========
+
+.. versionadded:: 1.23
+    The ``block`` test was added in Twig 1.23.
+
+``block`` checks if a block is defined in the current context of the template.
+
+.. code-block:: jinja
+
+    {% if 'title' is block %}
```

Comment:
@hason ``Twig_Template::hasBlock`` does not take inheritance into account, which precisely defeats the use case for this test: if it does not take inheritance into account, there is no reason to do this check at runtime; the developer can know it when writing the template and one of the branches of your ``if`` will always be dead.